### PR TITLE
Reorganize modules according to ARCHITECTURE.md

### PR DESCRIPTION
**Pull request by cgeorgii**

### CODE REVIEW

File: spec/Tagger/Repository/Content/InMemory.hs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module InMemoryContentRepository where
+module Tagger.Repository.Content.InMemory (inMemoryContentRepository) where
```
Comment (marcosh) [inline spec/Tagger/Repository/Content/InMemory.hs:1]: I'm not sure I understand the rationale beyond this move.

`Tagger` is the namespace reserved for the domain of the application.
Implementation of repositories are not part of the domain, so they do not belong there.
I would understand a move to `Infrastructure/Persistence/InMemoryContentRepository.hs`.

Moreover, why move `Content` to the namespace? It is not coherent with the rest of the structure of the project

File: src/Infrastructure/Persistence/PostgresUserRepository.hs [hunk 2]
```diff
@@ -17,7 +17,9 @@ import Infrastructure.Persistence.Serializer (serializeUser, unserializeUser)
 import Tagger.EncryptedPassword (EncryptedPassword)
 import Tagger.Id (Id (Id))
 import Tagger.User (User (User))
-import Tagger.UserRepository (UserRepository (..))
+import Tagger.Repository.User (UserRepository (..))
+
+type Ctx = ExceptT UserRepositoryError IO
```
Comment (marcosh) [inline src/Infrastructure/Persistence/PostgresUserRepository.hs:22]: I mentioned this also in the other PR. I don't consider such a type alias useful

File: src/Infrastructure/JSONWebKey.hs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 
-module Tagger.JSONWebKey (setup, JWK) where
+module Infrastructure.JSONWebKey (setup, JWK) where
```
Comment (marcosh) [inline src/Infrastructure/JSONWebKey.hs:3]: this makes sense to be in the infrastructure layer, good catch

File: src/Tagger/Repository/Content.hs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE RankNTypes #-}
 
-module Tagger.ContentRepository where
+module Tagger.Repository.Content(ContentRepository (..), hoist) where
```
Comment (marcosh) [inline src/Tagger/Repository/Content.hs:4]: why do you think it's better to expose explicitly all the things defined in the module?

**Review by marcosh — CHANGES_REQUESTED** [review]

File: src/Tagger/Repository/Content.hs (hunk 4 above)
Comment (cgeorgii) [reply src/Tagger/Repository/Content.hs:4]: Being explicit enforces separation of concerns and is helpful when projects grow. Having an explicit export list also helps with keeping the API of the module nice and clean by allowing to hide constructors helpers and other internals that shouldn't be exposed.

I will actually create a ticket to make all lists explicit.

File: src/Infrastructure/Persistence/PostgresUserRepository.hs (hunk 2 above)
Comment (cgeorgii) [reply src/Infrastructure/Persistence/PostgresUserRepository.hs:22]: Yeah, this was a mistake. I removed it from the signatures but the declaration remained.

File: spec/Tagger/Repository/Content/InMemory.hs (hunk 1 above)
Comment (cgeorgii) [reply spec/Tagger/Repository/Content/InMemory.hs:1]: The idea was to have an abstract representation of the repository and then specific concrete implementations:

- `Tagger.Repository.Content` (the abstract repository)
- `Tagger.Repository.Content.InMemory` (concrete implementation)
- `Tagger.Repository.Content.Postgres`(concrete implementation)

But maybe it makes more sense to put it in the Application layer?

- `Api.Repository.Content.InMemory`
- `Api.Repository.Content.Postgres`

Comment (marcosh) [reply spec/Tagger/Repository/Content/InMemory.hs:1]: concrete implementations are part of the infrastructure

**Pull request closed without merging**